core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -403,25 +403,37 @@ class DAGScheduler(
     parents.toList
   }
 
-  /** Find ancestor shuffle dependencies that are not registered in shuffleToMapStage yet */
-  private def getAncestorShuffleDependencies(rdd: RDD[_]): Stack[ShuffleDependency[_, _, _]] = {
-    val parents = new Stack[ShuffleDependency[_, _, _]]
+  /**
+   * Find ancestor shuffle dependencies that are not registered in shuffleToMapStage yet,
+   * in topological order to ensure building ancestor stages first.
```

Comment:
Nit: `This is done in topological order to ....` is better?
Also, I think we need to add comments for the reason why this topological order is needed here.